adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java

```diff
@@ -510,7 +470,41 @@ private void onLinkDetach(final ProtonReceiver receiver) {
             }
 
         }
-        return result;
+        return result.compose(address -> {
+            return validateAddress(address, authenticatedDevice);
+        });
+    }
+
+    private Future<Void> validateAddress(final ResourceIdentifier address, final Device authenticatedDevice) {
+        final Future<Void> addressCheck = Future.future();
+        if (address == null) {
+            addressCheck.complete();
```

Comment:
this is only here for the case that the device didn't provide an address at all. IMHO this should be handled in *validateEndpoint* already.